app/entry.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+require('./scss/reset.scss');
+require('./scss/main.scss');
+
+const angular = require('angular');
+const cowsay = require('cowsay-browser');
+
+const cowsayApp = angular.module('cowsayApp', []);
+
+cowsayApp.controller('CowsayController', ['$log', CowsayController]);
+
+function CowsayController($log) {
+  $log.debug('CowsayCont');
+
+  this.title = 'Geoff\'s Cowville!';
+  this.history = [];
+  cowsay.list( (err, cowfiles) => {
+    this.cowfiles = cowfiles;
+    this.current = 'dragon';
+  });
+
+  this.speak = function(input) {
+    $log.debug('cowsayCtrl.speak()');
+    return cowsay.say({
+      text: input || 'Whooooooshhh!',
+      f: this.current
+    });
+  };
+
+  this.logger = function(input) {
+    $log.debug('cowsayCtrl.logger()');
+    $log.log(input);
+  };
+
+  this.save = function() {
+    $log.debug('cowsayCtrl.save()', this.text);
+    if(this.text) {
+      this.history.push({
+        text: this.text,
+        f: this.current
+      });
+    }
+  };
+
+  this.last = function() {
```

Comment:
I really like how you put this in a function that always renders the last item. It makes the history handling really clean.